internal/pkg/cli/deploy/workload.go

```diff
@@ -486,6 +491,26 @@ func (d *workloadDeployer) pushAddonsTemplateToS3Bucket() (string, error) {
 }
 
 func (d *workloadDeployer) runtimeConfig(in *StackRuntimeConfiguration) (*stack.RuntimeConfig, error) {
+	out, err := d.convertRuntimeConfig(in)
+	if err != nil {
+		return nil, err
+	}
+	// Assign defaults if not set.
+	if len(out.CustomResourcesURL) == 0 {
+		// Populate the custom resource paths that copilot should upload the resources to.
+		crs, err := d.customResources(d.templateFS)
+		if err != nil {
+			return nil, err
+		}
+		out.CustomResourcesURL = make(map[string]string)
```

Comment:
It feels odd to me to check if `len(out.CustomResourcesURL) == 0` and if so initialize `out.CustomResourcesURL` again. Isn't it empty already?

Reply:
ooh yea, the problem is that if `out.CustomResourcesURL` is nil then we can't write `out.CustomResourcesURL[cr.Name()] = ...`

Like I have to initialize the map if it's nil if that makes sense.